### PR TITLE
GODRIVER-1963 remove dropDatabase result

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

File: x/mongo/driver/operation/drop_database.go [hunk 1]
```diff
@@ -31,66 +30,31 @@ type DropDatabase struct {
...
-
 // Execute runs this operations and returns an error if the operaiton did not execute successfully.
 func (dd *DropDatabase) Execute(ctx context.Context) error {
 	if dd.deployment == nil {
 		return errors.New("the DropDatabase operation must have a Deployment set before Execute can be called")
 	}
 
 	return driver.Operation{
-		CommandFn:         dd.command,
-		ProcessResponseFn: dd.processResponse,
```
Comment (benjirewis) [inline x/mongo/driver/operation/drop_database.go:84]: What happens if there's an error from `DropDatabase()`? Will the error not be returned now that there's no passed-in `ProcessResponseFn` when creating/executing the `Operation`?

Comment (iwysiu) [reply x/mongo/driver/operation/drop_database.go:84]: No, the only errors that ProcessResponseFn pass back are errors for the actual processing. Operation.decodeResult will separate out server errors before that, and those will get returned

Comment (benjirewis) [reply x/mongo/driver/operation/drop_database.go:84]: Ah that makes sense. Sounds good!

**Review by benjirewis — APPROVED** [review]
LGTM!

**Review by matthewdale — APPROVED** [review]
👍

**Pull request merged**